src/s3fs.cpp

```diff
@@ -4668,6 +4703,11 @@ static int my_fuse_opt_proc(void* data, const char* arg, int key, struct fuse_ar
             memset(&stbuf, 0, sizeof stbuf);
             set_mountpoint_attribute(stbuf);
 #else
+            // check stale mountpoint
+            if(stat(arg, &stbuf) == -1 && errno == ENOTCONN){
+                S3FS_PRN_WARN("MOUNTPOINT: directory %s is stale, try to unmount it.", mountpoint.c_str());
+                do_umount(mountpoint, true);
```

Comment:
I think it should check the return value of this function, and if it fails, call `S3FS_PRN_EXIT` and `return -1`.

Reply:
Thanks, I fixed it.